### DISCUSSION TITLE
next-auth writing duplicate id field to user in mongo db

### DISCUSSION THREAD

**Discussion by dubbajubba** [nextauthjs/next-auth#4071] [Help] [answered]
Hi there, and thanks for making this amazing library!

I'm using next-auth with a mongo db, and it works great, but I notice that some of my user documents have a duplicate id field. So I end up with the mongo db's _id field and an extra id field with the same value. I was wondering if there is a way to configure next-auth to not write the duplicate id.

I was thinking I could fix this with the session callback, but I must be missing something because it is still occurring. Here are my callbacks:
<img width="612" alt="image" src="https://user-images.githubusercontent.com/33437206/155656689-092d287f-00ae-4ba7-a4d3-251a7a06c28e.png">

And here are my type definitions:
<img width="586" alt="image" src="https://user-images.githubusercontent.com/33437206/155656831-0e6ea830-5ed5-47c4-9e2e-77edb6f2d1e5.png">

And let me know if I need to provide more details, information, or code please. This is actually my first ever question posted on GitHub!

**Comment by dubbajubba** [comment, marked as answer]
It looks like my issue should be fixed once this is merged: https://github.com/nextauthjs/next-auth/pull/4082
Thanks!

**Reply by balazsorban44** [reply]
Now merged. Will be part of the new release, hopefully coming :soon::tm:

**Reply by dubbajubba** [reply]
Awesome! Thanks so much! I'll keep an eye out!
